os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/da_ble_scatternet/ble_tizenrt_scatternet_app.c

```diff
@@ -125,12 +138,17 @@ void ble_tizenrt_scatternet_handle_callback_msg(T_TIZENRT_APP_CALLBACK_MSG callb
         }
 			break;
 
-        case BLE_TIZENRT_SCANED_DEVICE_MSG:
+        case BLE_TIZENRT_SCANNED_DEVICE_MSG:
         {
             debug_print("\r\n[%s] Handle scanned_device msg", __FUNCTION__);
             da_ble_client_scanned_device *scanned_device = callback_msg.u.buf;
-            client_init_parm->da_ble_client_device_scanned_cb(scanned_device);
-            os_mem_free(scanned_device);
+            if(scanned_device)
+            {
+                client_init_parm->da_ble_client_device_scanned_cb(scanned_device);
```

Comment:
Is it guarantee that `client_init_parm` is always not NULL?
If not, sometimes it can be dereferencing.

Reply:
Calling this function indicate that the ble stack has already init and still on so client_init_param is not NULL.